infra/main.tf

```diff
@@ -0,0 +1,190 @@
+terraform {
+  required_providers {
+    google = {
+      source = "hashicorp/google"
+      version = "3.48.0"
+    }
+    kubernetes = {
+      source = "hashicorp/kubernetes"
+      version = "1.13.3"
+    }
+  }
+}
+
+variable "gcp_project" {}
+variable "gcp_region" {}
+variable "gcp_zone" {}
+variable "domain" {}
+
+provider "google" {
+  credentials = file("~/.hail/terraform_sa_key.json")
+
+  project = var.gcp_project
+  region  = var.gcp_region
+  zone    = var.gcp_zone
+}
+
+data "google_client_config" "provider" {}
+
+resource "google_project_service" "service_networking" {
+  service = "servicenetworking.googleapis.com"
+}
+
+resource "google_compute_network" "internal" {
+  name = "internal"
+}
+
+data "google_compute_subnetwork" "internal_default_region" {
+  name = "internal"
+  region = var.gcp_region
+  depends_on = [google_compute_network.internal]
+}
+
+resource "google_container_cluster" "vdc" {
+  name = "vdc"
+  location = var.gcp_zone
+  network = google_compute_network.internal.name
+
+  # We can't create a cluster with no node pool defined, but we want to only use
+  # separately managed node pools. So we create the smallest possible default
+  # node pool and immediately delete it.
+  remove_default_node_pool = true
+  initial_node_count       = 1
+
+  master_auth {
```

Comment:
this block appears optional, is it necessary? 

https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster

Reply:
I think it is necessary.  From the linked page:

 > If this block is provided and both username and password are empty, basic authentication will be disabled. This block also contains several computed attributes, documented below. If this block is not provided, GKE will generate a password for you with the username admin.

I don't want basic auth.